dev/io.openliberty.microprofile.openapi.2.0.internal/src/io/openliberty/microprofile/openapi20/utils/CloudUtils.java

```diff
@@ -0,0 +1,90 @@
+/*******************************************************************************
+ * Copyright (c) 2020 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package io.openliberty.microprofile.openapi20.utils;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+import javax.ws.rs.HttpMethod;
+import javax.ws.rs.core.HttpHeaders;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.ibm.websphere.ras.Tr;
+import com.ibm.websphere.ras.TraceComponent;
+import com.ibm.ws.ffdc.annotation.FFDCIgnore;
+
+public class CloudUtils {
+    private static final TraceComponent tc = Tr.register(CloudUtils.class);
+
+    /**
+     * The getVCAPHost method checks whether a "VCAP_APPLICATION" environment variable has been set.  In Cloud Foundry
+     * applications (where Bluemix runs) this will be set to the actual host that is visible to the user.  In that
+     * environment the VHost from Liberty is private and not accessible externally. If the environment variable is set,
+     * it extracts the host and returns it;.
+     * 
+     * @return String
+     *          The host specified in VCAP_APPLICATION or null if it is not set.
+     */
+    @FFDCIgnore(Exception.class)
+    public static String getVCAPHost() {
```

Comment:
Do you still want to support this?

Reply:
Good question. Do Liberty instances still run in Cloud Foundry environments?  If so, then I guess we need to continue to support it.

However, it does raise the question of what we need to do to ensure that the correct host/port are included in the `servers` section when OL is running in OpenShift... sitting behind a `service` or, indeed, a `route`.

Reply:
This is additional behaviour when running in CF. I have no idea whether it's still required. I know we specifically opted not to do anything additional for CF on the reactive messaging feature, for example.

Reply:
Ok... and we could certainly take that approach for OpenAPI since users are able to specify their own content for the `servers` section using a variety of mechanisms.

Reply:
Ok, I don't mind either way, just as long as we've thought about it and decided what to do, rather than accidentally releasing something which we then have to support forever.